test/NuGet.Core.Tests/NuGet.PackageManagement.Test/DependencyGraphSpecTests.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using NuGet.ProjectModel;
+using Xunit;
+
+namespace NuGet.Test
+{
+    public class DependencyGraphSpecTests
```

Comment:
`WithPackageSpecs` is supposed to have the same behavior as calling `WithReplacedSpec` multiple time, right?  Maybe it's worth having a test that validates that.

Reply:
They're little bit different. `WithReplacedSpec` keeps only last 1 restore command and all projects. But `WithPackageSpecs` has more than 1 restore.